tests/test_autonomy/test_cli/test_deploy/test_build/test_deployment.py

```diff
@@ -318,235 +203,227 @@ def test_kubernetes_build_dev(
-                ]
-            )
-            assert "AEA_PASSWORD" in env.keys()
-            assert env["AEA_PASSWORD"] == ETHEREUM_ENCRYPTION_PASSWORD
+        assert "acn" in docker_compose["services"]
+        assert "hardhat" in docker_compose["services"]
 
-    def test_kubernetes_build_password(
+    def test_build_dev_failures(
         self,
     ) -> None:
         """Run tests."""
-        keys_file = Path(ETHEREUM_ENCRYPTED_KEYS)
 
         with mock.patch("os.chown"):
             result = self.run_cli(
                 (
-                    str(keys_file),
+                    str(self.keys_file),
                     "--o",
                     str(self.t / DEFAULT_BUILD_FOLDER),
                     "--force",
-                    "--kubernetes",
-                    "--password",
-                    ETHEREUM_ENCRYPTION_PASSWORD,
+                    "--dev",
                     "--local",
                 )
             )
 
-        build_dir = self.t / DEFAULT_BUILD_FOLDER
-        assert result.exit_code == 0, f"{result.stdout_bytes}\n{result.stderr_bytes}"
-        assert build_dir.exists()
+        assert result.exit_code == 1, result.output
+        assert "Please provide proper value for --packages-dir" in result.output
 
-        build_dir = self.t / DEFAULT_BUILD_FOLDER
-        kubernetes_config_file = build_dir / "build.yaml"
 
-        assert result.exit_code == 0, f"{result.stdout_bytes}\n{result.stderr_bytes}"
-        assert build_dir.exists()
-        with open(kubernetes_config_file, "r", encoding="utf-8") as fp:
-            kubernetes_config = list(yaml.safe_load_all(fp))
+class TestKuebernetesBuild(BaseDeployBuildTest):
+    """Test kubernetes builds."""
 
-        for resource in kubernetes_config:
-            try:
-                agent_vars = {
-                    f["name"]: f["value"]
-                    for f in resource["spec"]["template"]["spec"]["containers"][1][
-                        "env"
-                    ]
-                }
-            except (KeyError, IndexError):
-                continue
+    @staticmethod
+    def load_kubernetes_config(
+        path: Path,
+    ) -> List[Dict]:
+        """Load kubernetes config."""
 
-            assert agent_vars["AEA_PASSWORD"] == ETHEREUM_ENCRYPTION_PASSWORD
+        with open(path / "build.yaml", "r") as fp:
+            return list(yaml.safe_load_all(fp))
 
-        assert all(
-            [
-                (
-                    build_dir
-                    / DEPLOYMENT_KEY_DIRECTORY
-                    / KUBERNETES_AGENT_KEY_NAME.format(agent_n=i)
-                ).exists()
-                for i in range(4)
-            ]
+    @staticmethod
+    def check_kubernetes_build(build_dir: Path) -> None:
+        """Check kubernetes build dir."""
+
+        build_tree = list(map(lambda x: x.name, build_dir.iterdir()))
+        assert any(
+            [child in build_tree for child in ["persistent_storage", "build.yaml"]]
         )
```

Comment:
either? should this not be `all(c in build_tree for c in ["persistent_storage", "build.yaml"])`?